sql/2020/03_Markup/pages_element_count_by_device.sql

```diff
@@ -0,0 +1,69 @@
+#standardSQL
+# pages element_count metrics grouped by device
+
+# helper to create percent fields
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+# returns all the data we need from _element_count
+CREATE TEMPORARY FUNCTION get_element_count_info(element_count_string STRING)
+RETURNS STRUCT<
+  count INT64,
+  contains_custom_element BOOL, 
+  contains_obsolete_element BOOL,
+  contains_details_element BOOL,
+  contains_summary_element BOOL
+> LANGUAGE js AS '''
+var result = {};
+try {
+    if (!element_count_string) return result;
+
+    var element_count = JSON.parse(element_count_string);
+
+    if (Array.isArray(element_count) || typeof element_count != 'object') return result;
+
+    result.count = Object.values(element_count).reduce((total, freq) => total + (parseInt(freq, 10) || 0), 0);
+    
+    result.contains_custom_element = Object.keys(element_count).filter(e => e.includes('-')).length > 0;
+    result.contains_details_element = Object.keys(element_count).filter(e => e ==='details').length > 0;
+    result.contains_summary_element = Object.keys(element_count).filter(e => e ==='summary').length > 0;
+
+    var obsoleteElements = new Set(["applet","acronym","bgsound","dir","frame","frameset","noframes","isindex","keygen","listing","menuitem","nextid","noembed","plaintext","rb","rtc","strike","xmp","basefont","big","blink","center","font","marquee","multicol","nobr","spacer","tt"]);
+
+    result.contains_obsolete_element = !!Object.keys(element_count).find(e => {
+        return obsoleteElements.has(e);
+    });
+
+} catch (e) {}
+return result;
+''';
+
+SELECT
+  client,
+  COUNT(0) AS total,
+
+  CAST(ROUND(AVG(element_count_info.count)) AS INT64) AS elements_avg, # not sure what value this is? average elements per page? same for all device rows
+  MIN(element_count_info.count) AS elements_min, # y-axis min, same for all device rows
+  MAX(element_count_info.count) AS elements_max, # y-axis max, same for all device rows
```

Comment:
Should these be percentiles?

Reply:
They're not used. I think pulled in from last year. I'll drop them. A percentile of just element count would be of use and would cover all three. I'll add that.